### ISSUE TITLE
How can a TemplateResponse send and retrieve data?

### ISSUE THREAD

**Issue by tessavdheiden** [tiangolo/fastapi#1263]
How can I make a TemplateResponse that sends and gets data?

Initially, I want to have a html file that 1) allows users to input information, once this is done, 2) the information should be processed and the html file should be rendered with this information. 

Is it possible to do something like this:

1) items = []
output = templates.TemplateResponse("input.html", {"request": request, "items": items, "result": len(items)}
2) items = processing(output), templates.TemplateResponse("input.html", {"request": request, "items": items, "result": len(items)}

**Comment by jorgerpo** [comment]
In theory it should work
How did it go?

**Comment by tiangolo** [comment]
Yes, it could be done, but you would have to have custom logic in the template to show the form _or_ the result.

What I would suggest is that you have 2 independent templates, one for the form and one to render the data. ...actually the form doesn't even have to be a template, it could be plain HTML.

Then I would probably suggest having different URLs for the input data and for the results after submitting the result. But if you definitely need to have them in the same URL then you could still just have one `get` to show the HTML form and one `post` to show the rendered results.
